src/models/transferSettlement/facade.js

```diff
@@ -0,0 +1,305 @@
+                  .innerJoin('participantCurrency AS PC1', function () {
+                    this.on('PC1.currencyId', 'PC.currencyId')
+                      .andOn('PC1.participantId', 'PC.participantId')
+                      .andOn('PC1.ledgerAccountTypeId', 'M.settlementAccountTypeId')
+                  })
+                  .where(function () {
+                    this.where({ 'TP.transferId': transferId })
+                    this.andWhere(function () {
+                      this.andWhere({ 'G.name': 'GROSS' })
+                    })
+                  })
+              })
+          })
+          .transacting(trx)
+
+        // Insert a new status for the transfer.
+        const transferStateChange = [
+          {
+            transferId: transferId,
+            transferStateId: 'SETTLED',
+            reason: 'Gross settlement process'
+          }
+        ]
+
+        await knex('transferStateChange').insert(transferStateChange)
+          .transacting(trx)
+
+        // Update the positions
+        await knex('participantPosition AS PP')
+          .update({ value: knex.raw('?? - ??', ['PP.value', 'TR.amount']) })
+          .innerJoin(function () {
+            this.from('transferParticipant AS TP')
+              .select('PC.participantCurrencyId', 'TP.Amount')
+              .innerJoin('participantCurrency AS PC', 'TP.participantCurrencyId', 'PC.participantCurrencyId')
+              .innerJoin('settlementModel AS M', 'M.ledgerAccountTypeId', 'PC.ledgerAccountTypeId')
+              .innerJoin('settlementGranularity AS G', 'M.settlementGranularityId', 'G.settlementGranularityId')
+              .where(function () {
+                this.where({ 'TP.transferId': transferId })
+                this.andWhere(function () {
+                  this.andWhere({ 'G.name': 'GROSS' })
+                })
+              })
+              .union(function () {
+                this.select('PC1.participantCurrencyId', 'TP.amount')
+                  .from('transferParticipant AS TP')
+                  .innerJoin('participantCurrency AS PC', 'TP.participantCurrencyId', 'PC.participantCurrencyId')
+                  .innerJoin('settlementModel AS M', 'M.ledgerAccountTypeId', 'PC.ledgerAccountTypeId')
+                  .innerJoin('settlementGranularity AS G', 'M.settlementGranularityId', 'G.settlementGranularityId')
+                  .innerJoin('participantCurrency AS PC1', function () {
+                    this.on('PC1.currencyId', 'PC.currencyId')
+                      .andOn('PC1.participantId', 'PC.participantId')
+                      .andOn('PC1.ledgerAccountTypeId', 'M.settlementAccountTypeId')
+                  })
+                  .where(function () {
+                    this.where({ 'TP.transferId': transferId })
+                    this.andWhere(function () {
+                      this.andWhere({ 'G.name': 'GROSS' })
```

Comment:
'GROSS' should be set using an ENUM from central-services-shared.

Reply:
As discussed with @elnyry-sam-k: flag this issue for a future story If appropriate ENUMS are not already available for this.